plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/AmazonS3WithCredentials.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.s3;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.s3.AmazonS3;
+
+import org.opensearch.common.Nullable;
+
+/**
+ * The holder of the AmazonS3 and AWSCredentialsProvider
+ */
+final class AmazonS3WithCredentials {
+    private final AmazonS3 client;
+    private final AWSCredentialsProvider credentials;
+
+    private AmazonS3WithCredentials(final AmazonS3 client, @Nullable final AWSCredentialsProvider credentials) {
+        this.client = client;
+        this.credentials = credentials;
+    }
+
+    AmazonS3 getClient() {
+        return client;
+    }
+
+    AWSCredentialsProvider getCredentials() {
+        return credentials;
+    }
+
+    AmazonS3 client() {
+        return client;
+    }
+
+    AWSCredentialsProvider credentials() {
+        return credentials;
+    }
```

Comment:
Did you mean to double the getters here (with and without the `get` prefix)? Probably don't need both.

Reply:
Oh sorry, I missed that, you are right - no need for getters, thanks!